### ISSUE TITLE
Cygwin issue: Segmentation fault when generating test-full-cxx-output.inc

### ISSUE THREAD

**Issue by Sean-MA** [protobuf-c/protobuf-c#483]
Hello,

I am trying to compile the protobuc-c on cygwin.
I use: _./autogen.sh && ./configure && ./make_
Everything looks normal when I run _./autogen.sh_ and _./configure_.
The terminal outputs when I run _./make_:

_...
CXXLD    t/generated-code2/cxx-generate-packed-data.exe
GEN      t/generated-code2/test-full-cxx-output.inc
/bin/sh: line 1:  5226 Segmentation fault      (core dumped) ./t/generated-code2/cxx-generate-packed-data.exe > ./t/generated-code2/test-full-cxx-output.inc
make: *** [Makefile:2719: t/generated-code2/test-full-cxx-output.inc] Error 139_

_GEN      t/generated-code2/test-full-cxx-output.inc_ is to generate _test-full-cxx-output.inc_ by using _cxx-generate-packed-data.exe_.
I ran _./cxx-generate-packed-data.exe > test.inc_, and I still have a segmentation fault. 
So, to my understanding, it seems there is something wrong with the link stage (i.e., CXXLD)? 
Any help? Thanks in advance.

- The version of protobuf-c is 1.4.0
- The version of cygwin is mintty 3.5.0 (x86_64-pc-cygwin) [Windows 18363]
- The version of protobuf is 3.17.3

**Comment by Sean-MA** [comment]
However, it works in protobuf-c 1.3.3.

**Comment by adrianreber** [comment]
I also see segfaults building protobuf-c 1.40 against protobuf 3.18.1 on Fedora:
```
cxx-generate-pa[617783]: segfault at 20 ip 00007f112ad1f481 sp 00007ffcce498ea0 error 4 in libprotobuf.so.29.0.1[7f112ace8000+1b2000]
Code: fd e8 a3 40 ff ff 48 8b 15 84 71 21 00 48 98 83 3c 82 0a 75 34 48 8b 6c 24 08 4c 89 e7 66 81 65 09 00 0f 49 8b 04 24 48 8b 33 <ff> 50 20 48 89 45 00 48 8b 44 24 08 80 60 0a f0 48 8b 00 e9 9b fe
```

**Comment by tbvdm** [comment]
I'm able to build protobuf-c 1.4.0 and 1.4.1 on Cygwin with `CXXFLAGS=-std=c++17`.

**Comment by edmonds** [comment]
This might be fixed in 1.5.0, please reopen if that's not the case.
